std/file.d

```diff
@@ -2318,7 +2318,9 @@ private bool ensureDirExists(in char[] pathname)
  * Throws: $(D FileException) on error.
  */
 
-void mkdirRecurse(in char[] pathname)
+void mkdirRecurse(R)(R pathname)
+if (isInputRange!R && !isInfinite!R && isSomeChar!(ElementEncodingType!R) &&
+    !isConvertibleToString!R)
```

Comment:
Oh good catch! I went with changing `ensureDirExists`, but I don't have a strong feeling about this.